### PR TITLE
forbid creating price of a different type in createPrice procedure

### PR DESCRIPTION
**Pull request by angihe93**
## What Does this PR Do?

---
## Summary by cubic
Prevent creating a price with a different type than existing prices for the same product in createPrice. This enforces consistent price types per product and returns a FORBIDDEN error if a mismatch is attempted.

<sup>Written for commit f6332418f58b6c323513c6a8e3c9b9f0d89822fe. Summary will update automatically on new commits.</sup>

### CODE REVIEW

**Review by agreea — CHANGES_REQUESTED** [review]
The code here is fine but the number of checks makes me worried that we now have too much logic that is going untested in this endpoint. If we move it all into a single db transaction (in `utils/pricingModels.ts`) and then create test coverage for it, that would be a nice improvement. Esp since we keep finding edge cases for when to allow / disallow creating prices - it would be good to formalize it with test coverage.

File: platform/flowglad-next/src/utils/pricingModel.ts [hunk 1]
```diff
@@ -52,6 +57,94 @@ export const createPrice = async (
...
+  const product = await selectProductById(
+    price.productId,
+    transaction
+  )
+  if (!product) {
+    throw new TRPCError({
+      code: 'NOT_FOUND',
+      message: `Product not found with id ${price.productId}`,
+    })
+  }
```
Comment (agreea) [inline platform/flowglad-next/src/utils/pricingModel.ts]: Don't need to throw here as findById will only return if there is a product found. Otherwise it will throw internally

**Review by agreea — APPROVED** [review]
Good to ship pending some small suggestions re: `TRPCError` conditions that will never be hit because `findById` throws an error internally if the record with that id (organization, customer, etc) is not found
